bftengine/src/bftengine/ReadOnlyReplica.hpp

```diff
@@ -43,7 +43,8 @@ class ReadOnlyReplica : public ReplicaForStateTransfer {
 
   template <typename T>
   void messageHandler(std::unique_ptr<MessageBase> msg) {
-    auto trueTypeObj = std::make_unique<T>(msg.release());
+    auto trueTypeObj = std::make_unique<T>(msg.get());
+    delete msg.release();
```

Comment:
The proper way to do this is:
`msg.reset()`
Same for all other places where you
`delete smart_ptr.release();`
(I won't comment anywhere else)

Reply:
Agree